### PR TITLE
Address fields for Stripe payment

### PR DESCRIPTION
**Pull request by pattihis**
### 🎫 Ticket

[ETP-934]

### 🗒️ Description

If Stripe is our payment gateway, and if more payment methods than simple Credit Cards are available, then we render a form to collect users address and we pass that to Stripe as billing and shipping data. 

### ✔️ Checklist
- [x] Ran `npm run changelog` to add changelog file(s).
- [ ] Code is covered by **NEW** `wpunit` or `integration` tests.
- [x] Code is covered by **EXISTING** `wpunit` or `integration` tests.
- [ ] Are all the **required** tests passing?
- [x] Automated code review comments are addressed.
- [ ] Have you added Artifacts?
- [x] Check the base branch for your PR.
- [x] Add your PR to the project board for the release.

[ETP-934]: https://stellarwp.atlassian.net/browse/ETP-934?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/resources/js/commerce/gateway/stripe/checkout.js [hunk 1]
```diff
@@ -604,6 +668,7 @@ tribe.tickets.commerce.gateway.stripe.checkout = {};
 	 * @since 5.3.0
 	 */
 	obj.bindEvents = () => {
+		$( obj.selectors.renderButton ).on( 'click', obj.renderPayment );
```
Comment (borkweb) [inline src/resources/js/commerce/gateway/stripe/checkout.js:671]: I know that this is mimicking the line below it, however, it is good practice to bind to the document or a container in the event that the item is injected into the page via lazy loading.

```suggestion
		$( document ).on( 'click', obj.selectors.renderButton, obj.renderPayment );
```

File: src/resources/postcss/tickets-commerce/_checkout.pcss [hunk 2]
```diff
@@ -322,6 +322,20 @@
...
+		}
+	}
+
+	.tribe-tickets__commerce-checkout-purchaser-info-form-field.error {
+		border: 1px solid red!important;
+	}
+
+	.tribe-tickets__commerce-checkout-address-wrapper {
+		display: grid;
+		grid-template-columns: 1fr 1fr;
```
Comment (borkweb) [inline src/resources/postcss/tickets-commerce/_checkout.pcss:337]: I will _always_ read fr fr as "for real, for real" - hahaha 🤪

File: src/views/v2/commerce/checkout/gateways.php [hunk 3]
```diff
@@ -26,8 +26,13 @@
...
 
-// Bail if user needs to login, the cart is empty or if there are no active gateways.
-if ( $must_login || empty( $items ) || ! tribe_is_truthy( $gateways_active ) ) {
+// Bail if the cart is empty or if there are no active gateways.
+if ( empty( $items ) || ! tribe_is_truthy( $gateways_active ) ) {
+	return;
+}
+
+// Bail if user needs to login, but is not logged in.
+if ( ! is_user_logged_in() && $must_login ) {
```
Comment (borkweb) [inline src/views/v2/commerce/checkout/gateways.php:35]: A micro optimization, but let's move the more expensive item second:

```suggestion
if ( $must_login && ! is_user_logged_in() ) {
```

File: src/views/v2/commerce/checkout/purchaser-info.php [hunk 4]
```diff
@@ -18,13 +18,45 @@
...
  */
 
-if ( is_user_logged_in() || $must_login || empty( $items ) ) {
+// Bail if the cart is empty.
+if ( empty( $items ) ) {
+	return;
+}
+
+// Bail if user needs to login, but is not logged in.
+if ( ! is_user_logged_in() && $must_login ) {
```
Comment (borkweb) [inline src/views/v2/commerce/checkout/purchaser-info.php:27]:
```suggestion
if ( $must_login && ! is_user_logged_in() ) {
```

File: src/views/v2/commerce/checkout/purchaser-info.php [hunk 5]
```diff
@@ -18,13 +18,45 @@
...
 	return;
 }
-?>
 
+$info_title   = __( 'Purchaser info', 'event-tickets' );
+$show_address = false;
+foreach ( $gateways as $gateway ) {
+	// Check if Stripe is active and enabled.
+	if ( 'stripe' === $gateway::get_key() && $gateway::is_enabled() && $gateway::is_active() ) {
```
Comment (borkweb) [inline src/views/v2/commerce/checkout/purchaser-info.php:35]: Our ~standard~ general practice is to split to multi-lines when the conditionals exceed two for readability:

_Note: I thought I had put that in our standards, but I guess that is something that we don't draw a hard line on, just encourage for readability._

```suggestion
	if (
		'stripe' === $gateway::get_key()
		&& $gateway::is_enabled()
		&& $gateway::is_active()
	) {
```

File: src/views/v2/commerce/checkout/purchaser-info.php [hunk 6]
```diff
@@ -18,13 +18,45 @@
...
+		$payment_methods = ( new TEC\Tickets\Commerce\Gateways\Stripe\Merchant() )->get_payment_method_types();
+		// If more than one payment method, or if only one but not a card, we need to show the address fields.
+		if ( 1 < count( $payment_methods ) || ( 1 === count( $payment_methods ) && 'card' !== $payment_methods[0] ) ) {
```
Comment (borkweb) [inline src/views/v2/commerce/checkout/purchaser-info.php:38]:
```suggestion
		if (
			1 < count( $payment_methods )
			|| (
				1 === count( $payment_methods )
				&& 'card' !== $payment_methods[0]
			)
		) {
```

**Review by borkweb — CHANGES_REQUESTED** [review]
Looking good! A couple of tweaks

File: src/admin-views/settings/tickets-commerce/stripe/modal/signup-complete/content.php [hunk 7]
```diff
@@ -55,6 +55,18 @@
 		?>
 	</div>
 
+	<div class="tec-tickets__admin-settings-tickets-commerce-stripe-modal-content-section">
+		<strong><?php esc_html_e( 'Billing Fields', 'event-tickets' ); ?></strong> &mdash;
+		<?php
+		printf(
+			// Translators: %1$s: opening `a` tag to the knowledge base article. %2$s: closing `a` tag.
+			esc_html__( 'If you use certain payment providers with Stripe including Afterpay, Klarna or Clearpay additional Billing Fields will be displayed at checkout as required by Stripe. %1$sLearn More%2$s.', 'event-tickets' ),
```
Comment (borkweb) [inline src/admin-views/settings/tickets-commerce/stripe/modal/signup-complete/content.php:63]: Small grammar fixes I missed before

```suggestion
			esc_html__( 'If you use certain payment providers with Stripe, including Afterpay, Klarna, or Clearpay, additional Billing Fields will be displayed at checkout as required by Stripe. %1$sLearn More%2$s.', 'event-tickets' ),
```

File: src/Tickets/Commerce/Gateways/Stripe/Settings.php [hunk 8]
```diff
@@ -203,7 +203,7 @@ public function get_settings() {
 
 		$payment_methods_tooltip = sprintf(
 		// Translators: %1$s: Opening `<span>` tag. %2$s: The currency name. %3$s: Closing `</span>` tag. %4$s: Opening `<a>` tag for Stripe link. %5$s: Closing `</a>` tag.
-			__( '%1$sPayment methods available for %2$s%3$s.<br /><br /> The payment methods listed here are dependent on the currency selected for Tickets Commerce and the currency each payment method support. You can review the payment methods and their availablity for each currency on %4$sStripe\'s documentation%5$s.<br /><br />', 'event-tickets' ),
+			__( '%1$sPayment methods available for %2$s%3$s.<br /><br /> The payment methods listed here are dependent on the currency selected for Tickets Commerce and the currency each payment method support. If you select a payment provider outside of Credit Cards, additional billing information fields will display on checkout as required by Stripe. You can review the payment methods and their availablity for each currency on %4$sStripe\'s documentation%5$s.<br /><br />', 'event-tickets' ),
```
Comment (borkweb) [inline src/Tickets/Commerce/Gateways/Stripe/Settings.php:206]:
```suggestion
			__( '%1$sPayment methods available for %2$s%3$s.<br /><br /> The payment methods listed here depend on the currency selected for Tickets Commerce and the currency supported by each payment method. If you select a payment provider other than Credit Cards, additional billing information fields will be displayed at checkout as required by Stripe. You can review the payment methods and their availablity for each currency on %4$sStripe\'s documentation%5$s.<br /><br />', 'event-tickets' ),
```

**Review by borkweb — CHANGES_REQUESTED** [review]
I missed some small grammar changes when I did the review last night 🥼

File: src/resources/postcss/tickets-commerce/_checkout.pcss [hunk 9]
```diff
@@ -334,6 +352,15 @@
 		width: 100%;
 	}
 
+	.tribe-tickets__form-field-input-wrapper {
+
+		input::placeholder {
+  		color: var(--tec-color-border-tertiary);
+  		font-size: var(--tec-font-size-2);
+  		opacity: 1;
+		}
```
Comment (codingmusician) [inline src/resources/postcss/tickets-commerce/_checkout.pcss:361]:
```suggestion
		input::placeholder {
  			color: var(--tec-color-border-tertiary);
  			font-size: var(--tec-font-size-2);
  			opacity: 1;
		}
```

**Review by codingmusician — DISMISSED** [review]

**Review by borkweb — DISMISSED** [review]

**Review by dpanta94 — APPROVED** [review]

**Pull request merged**